web/meetup-facilitator/src/pages/Activities.jsx

```diff
@@ -0,0 +1,62 @@
+import { createContext, useEffect, useRef, useState } from "react"
+import Card from "../components/Card"
+import PopUp from "../components/PopUp";
+
+export  const activitiesContext = createContext();
+
+
+function Activities() {
+
+    const [activities, setActivities] = useState({});
+    const [showPopUp, setShowPopUp] = useState(false);
+    const [targetCategory, setTargetCategory] = useState([]);
+    const pendingPrefs = useRef({});
+
+    const subactivities = Object.values(activities);
+
+    const mainCategories = Object.keys(activities).map((category, i) => {
+
+        return (
+            <li key={i}>
+                <activitiesContext.Provider value={{setShowPopUp, setTargetCategory, showPopUp}}>
+                    <Card activity={category} subcategories={Object.keys(subactivities[i])}/>
```

Comment:
Let's pass down `subactivities[i]` instead of just the keys. This way we can use the Google activity ids as ids for each of the inputs which will be more standard than using the actual names (cuz names can have spaces, apostrophes, and other special characters, and we don't know how that's gonna play out in the DOM tree when the browser is trying to search for the id).